src/requests/system_info.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
```

Comment:
looks like a shared header

Reply:
What is shared header? And what exactly does make it looking like this?
If you mean its location in src dir? 
Frankly I don't see why it should be located somewhere else.